web/src/components/ChatConversation.tsx

```diff
@@ -29,10 +33,15 @@ type ChatConversationProps = {
   className?: string
 }
 
+const TYPING_INDICATOR_TIMEOUT = 60000
+
 const ChatConversation = ({ messages, hasError, className }: ChatConversationProps): ReactElement => {
   const { t } = useTranslation('chat')
   const [messagesCount, setMessagesCount] = useState(0)
+  const [showTypingIndicator, setShowTypingIndicator] = useState(false)
   const messagesEndRef = useRef<HTMLDivElement>(null)
+  const lastUserMessage = messages[messages.length - 1]
+  const beforelastUserMessage = messages[messages.length - 2]
```

Comment:
:wrench: The naming is misleading here, those are not necessarily user messages.
```suggestion
  const lastMessage = messages[messages.length - 1]
  const secondLastMessage = messages[messages.length - 2]
```
Perhaps even better would be the following to have a speaking variable:
```suggestion
  const showTypingIndicator = messages[messages.length - 1]?.userIsAuthor || messages[messages.length - 2]?.userIsAuthor
```